ios/RPCModule.swift

```diff
@@ -11,169 +11,144 @@ import React
         }
       } else {
         NSLog("could not decode b64 content to save backup wallet.")
       }
   }
 
   func saveBackgroundFile(_ data: String) {
-      let paths = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true)
-      if let documentsDirectory = paths.first {
-        let fileName = "\(documentsDirectory)/background.json"
-        do {
-            try data.write(toFile: fileName, atomically: true, encoding: .utf8)
-        } catch {
-            NSLog("Error save background file \(error.localizedDescription)")
-        }
-      } else {
-        NSLog("Error save background file")
+      let fileName = get_filename("background.json")
+      do {
+          try data.write(toFile: fileName, atomically: true, encoding: .utf8)
+      } catch {
+          NSLog("Error save background file \(error.localizedDescription)")
       }
   }
 
   // old way to read the wallet file -> Encoded Utf8 String
   func readWalletUtf8String() -> String? {
-      let paths = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true)
-      if let documentsDirectory = paths.first {
-        let fileName = "\(documentsDirectory)/wallet.dat.txt"
-        do {
-            let content = try String(contentsOfFile: fileName, encoding: .utf8)
-            //NSLog("load encoded utf8 string wallet \(content)")
-            return content
-        } catch {
-            NSLog("Error reading wallet \(error.localizedDescription)")
-            return nil
-        }
-      } else {
-        NSLog("Error reading wallet")
-        return nil
+      let fileName = get_filename("wallet.dat.txt")
+      do {
+          let content = try String(contentsOfFile: fileName, encoding: .utf8)
+          //NSLog("load encoded utf8 string wallet \(content)")
+          return content
+      } catch {
+          NSLog("Error reading wallet \(error.localizedDescription)")
+          return nil
       }
   }
 
   // new way to read the wallet file -> Decoded Data
   func readWallet() -> Data? {
-      let paths = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true)
-      if let documentsDirectory = paths.first {
-        let fileName = "\(documentsDirectory)/wallet.dat.txt"
-        do {
-            let content = try Data(contentsOf: URL(fileURLWithPath: fileName))
-            //NSLog("load decoded data wallet \(content)")
-            return content
-        } catch {
-            NSLog("Error reading wallet \(error.localizedDescription)")
-            return nil
-        }
-      } else {
-        NSLog("Error reading wallet")
-        return nil
+      let fileName = get_filename("wallet.dat.txt")
+      do {
+          let content = try Data(contentsOf: URL(fileURLWithPath: fileName))
+          //NSLog("load decoded data wallet \(content)")
+          return content
+      } catch {
+          NSLog("Error reading wallet \(error.localizedDescription)")
+          return nil
```

Comment:
I will look into an explicit error to be thrown here.  It seems like the interfaces to react might use "resolve" but I am not clear on this, yet.

Reply:
The functions with tag `@objc` have resolve & reject parameter and they need to finish with one or the other executed.